zanata-war/src/main/java/org/zanata/limits/RateLimitingProcessor.java

```diff
@@ -48,16 +58,20 @@ private void process(String key, HttpResponse response, Runnable taskToRun)
                         "{} has too many concurrent requests. Returning status 429",
                         key);
             }
+
+            String clientIdentifier = "";
+            if(key.getType().equals(RateLimiterToken.TYPE.API_KEY)) {
+                clientIdentifier = "API key";
+            } else  {
+                clientIdentifier = key.getValue();
+            }
+
             String errorMessage =
                     String.format(
-                            "Too many concurrent requests for this user (maximum is %d)",
+                            "Too many concurrent requests for client '%s' (maximum is %d)",
```

Comment:
Probably best not to put "API key" in quotes here, or it will look like a username: `Too many concurrent requests for client 'API key'`.  Since you have access to the actual Token type here, you could customise the message properly: `username '%s`", `IP address '%s'` or `API key`.